### PR TITLE
More consistency changes in A.M.Q

### LINKED ISSUE #20859: Enable NetworkTimeout support in Azure.Monitor.Query
It was disabled because the feature was disabled in Azure.Core.

### LINKED ISSUE #23513: Implement service timeout for batches
We should pick the max timeout from all the queries and use that for the pipeline setting.

### LINKED ISSUE #24021: Azure Monitor Query - switch to new logs endpoint
Use https://api.monitor.azure.com as the default endpoint for LogsQueryClient

### PR DESCRIPTION
**Pull request by pakrym**
Based on VTeam review.

Rename MonitorQueryTimeRange to QueryTimeRange 
Change the default endpoint for logs client. Fixes #24021
Add Endpoint property to query client.
Follow ACR example and add ***Audience types and properties to options.
Use max timeout from batch for the overall request. Fixes #23513 Fixes #20859

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 1]
```diff
@@ -487,7 +464,80 @@ internal static QueryBody CreateQueryBody(string query, MonitorQueryTimeRange ti
...
+                    (timeout == null || requestTimeout.Value > timeout.Value))
+                {
+                    timeout = requestTimeout;
+                }
+            }
+
+            if (timeout != null)
+            {
+                message.NetworkTimeout = timeout.Value.Add(_networkTimeoutOffset);
+                message.Request.Headers.SetValue("prefer", $"wait={(int) timeout.Value.TotalSeconds}");
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:506]: Can we add this common header to [HttpHeader](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/HttpHeader.cs)?

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientAudience.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        /// <summary>
+        /// Initializes a new instance of <see cref="LogsQueryClientAudience"/> with a given value.
+        /// </summary>
+        public LogsQueryClientAudience(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        /// <summary> Azure Public Cloud. </summary>
+        public static LogsQueryClientAudience AzurePublicCloud { get; } = new("https://api.loganalytics.io");
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientAudience.cs:25]: Are there not any other clouds that support this service?

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientOptions.cs [hunk 3]
```diff
@@ -45,9 +45,9 @@ public enum ServiceVersion
         }
 
         /// <summary>
-        /// Gets or sets the authentication scope to use for authentication with Azure Active Directory. The default scope will be used if the property is null.
+        /// Gets or sets the audience to use for authentication with Azure Active Directory. The default scope will be used if the property is null.
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientOptions.cs:48]: mention the default is Azure public cloud?

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs [hunk 4]
```diff
@@ -43,6 +43,46 @@ public void CanSetServiceTimeout_Mocked()
...
+                ServerTimeout = TimeSpan.FromMinutes(2)
+            });
+            batch.AddQuery("wid", "tid", TimeSpan.FromDays(1), options: new LogsQueryOptions()
+            {
+                ServerTimeout = TimeSpan.FromMinutes(3)
+            });
+
+            Assert.ThrowsAsync<RequestFailedException>(() => client.QueryBatchAsync(batch));
+
+            Assert.AreEqual("wait=180", preferHeader);
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs:81]: why `180`?

**Review by christothes — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientAudience.cs (hunk 2 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClientAudience.cs:25]: We are still figuring out what audiences for those are :) I'll add them as soon as we know.

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs (hunk 4 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs:81]: 3 minutes :) Or are you saying I should add a comment?

Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs:81]: Even better than a comment would be to reference the value that enforces the 3 minutes. Is it defined somewhere or is it configured as an option? If it's a constant value defined by the service, could it ever change?

File: sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQuery.cs [hunk 5]
```diff
@@ -29,11 +29,11 @@ public LogsBatchQuery()
         /// string countQueryId = batch.AddQuery(
         ///     workspaceId,
         ///     &quot;AzureActivity | count&quot;,
-        ///     new MonitorQueryTimeRange(TimeSpan.FromDays(1)));
+        ///     new QueryTimeRange(TimeSpan.FromDays(1)));
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQuery.cs:32]: Didn't realize `<code>` tags on their own look worse.

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs (hunk 4 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs:81]: We take MAX of individual query timeouts so it's based on test data.

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:506]: I kinda hate that type :) I don't think it adds much value over a constant string tbh.

File: sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQuery.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQuery.cs:32]: The code looks the same, but it just gets its own Example header in the docs. That seems like the more common pattern when <code> tags are in a doc comment.

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs (hunk 4 above)
Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs:81]: sweet - should we calculate the value in the Assert then?

**Pull request merged**